src/GovUk.Education.ExploreEducationStatistics.Data.Services/ReleaseService.cs

```diff
@@ -73,9 +74,22 @@ private async Task<List<SubjectViewModel>> GetSubjects(Guid releaseVersionId, Li
 
             var releaseFiles = await QueryReleaseDataFiles(releaseVersionId)
                 .Where(rf => rf.File.SubjectId.HasValue
+                             && rf.ReleaseVersionId == releaseVersionId
                              && subjectsToInclude.Contains(rf.File.SubjectId.Value))
                 .ToListAsync();
 
+            // This check was added to protect against EES-5404
+            if (releaseSubjects.Count != releaseFiles.Count)
```

Comment:
This might be ok, but I was thinking that we'd do this check earlier than this, after assigning `releaseSubjects`.

The `subjectsToInclude` parameter is a list of subject id's found in the content database. See how it gets worked out in the method `GetPublishedSubjectIds`.

So based on this list of subject id's that we expect to exist from data in the content database, we get the subjects from statistics database and assign `releaseSubjects`.

If any subject in that list is not found we'd return an error.

Reply:
I've made the change. 

I will just highlight that this means that if someone changes the way `subjectsToInclude` is fetched for some reason, then  this check might be meaningless. But I don't see how or why that change would happen, and if it did it would be in the far future, so I think it's fine. I was a little tempted to change the variable name to make it explicit, but something like`contentDbSubjectsToInclude` seems a bit extreme.